### PR TITLE
Home 7

### PR DESCRIPTION
**Pull request by IgorPrihodko**

### CODE REVIEW

File: src/main/java/controller/InitServlet.java [hunk 1]
```diff
@@ -22,11 +22,19 @@
     @Override
     public void init() throws ServletException {
         User user = new User("test@test", "test", "admin");
+        User user1 = new User("111@111", "111", "user");
```
Comment (Sofasmile) [inline src/main/java/controller/InitServlet.java]: cs

File: src/main/java/controller/InitServlet.java [hunk 2]
```diff
@@ -22,11 +22,19 @@
...
+        User user2 = new User("igorprihodkoemail@gmail.com", "111", "user");
         userService.addUser(user);
+        userService.addUser(user1);
+        userService.addUser(user2);
         logger.warn("Add initial user " + user + " to DB");
 
         Product product = new Product("test", "test", 0.0);
+        Product product1 = new Product("test1", "test1", 0.1);
```
Comment (Sofasmile) [inline src/main/java/controller/InitServlet.java]: cs

File: src/main/java/controller/admin/AllProductServlet.java [hunk 3]
```diff
@@ -22,6 +22,6 @@
     protected void doGet(HttpServletRequest req, HttpServletResponse resp)
```
Comment (Sofasmile) [inline src/main/java/controller/admin/AllProductServlet.java:20]: move allProducts in method

File: src/main/java/controller/user/ProductForUsersServlet.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.List;
+
+@WebServlet(value = "/user/products")
+public class ProductForUsersServlet extends HttpServlet {
+
+    private static final ProductService productService = ProductServiceFactory.getInstance();
+    private List<Product> allProducts = productService.getAll();
```
Comment (Sofasmile) [inline src/main/java/controller/user/ProductForUsersServlet.java]: better move in the  method

File: src/main/java/model/User.java [hunk 5]
```diff
@@ -1,16 +1,23 @@
...
+import java.util.ArrayList;
+import java.util.List;
+
 public class User {
 
     private Long id;
     private String email;
     private String password;
     private String role;
+    private List<Order> orders;
```
Comment (Sofasmile) [inline src/main/java/model/User.java]: do we need this field here? and basket too?

File: src/main/java/service/mail/MailServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+            message.setRecipients(
+                    Message.RecipientType.TO,
+                    InternetAddress.parse(code.getUserEmail())
+            );
+            message.setSubject("Confirmation code");
+            message.setText("Enter this code - [" + code.getCode() + "] for confirmation");
+
+            Transport.send(message);
+
+            System.out.println("Done");
```
Comment (Sofasmile) [inline src/main/java/service/mail/MailServiceImpl.java]: sout???

File: src/main/java/service/mail/MailServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+
+        } catch (MessagingException e) {
+            e.printStackTrace();
```
Comment (Sofasmile) [inline src/main/java/service/mail/MailServiceImpl.java]: '_' ))

File: src/main/java/controller/InitServlet.java (hunk 1 above)
Comment (IgorPrihodko) [reply src/main/java/controller/InitServlet.java]: fixed

File: src/main/java/controller/InitServlet.java (hunk 2 above)
Comment (IgorPrihodko) [reply src/main/java/controller/InitServlet.java]: fixed

File: src/main/java/controller/admin/AllProductServlet.java (hunk 3 above)
Comment (IgorPrihodko) [reply src/main/java/controller/admin/AllProductServlet.java:20]: fixed

File: src/main/java/controller/user/ProductForUsersServlet.java (hunk 4 above)
Comment (IgorPrihodko) [reply src/main/java/controller/user/ProductForUsersServlet.java]: fixed

File: src/main/java/service/mail/MailServiceImpl.java (hunk 6 above)
Comment (IgorPrihodko) [reply src/main/java/service/mail/MailServiceImpl.java]: fixed

File: src/main/java/service/mail/MailServiceImpl.java (hunk 7 above)
Comment (IgorPrihodko) [reply src/main/java/service/mail/MailServiceImpl.java]: my fall) fixed

File: src/main/java/model/User.java (hunk 5 above)
Comment (IgorPrihodko) [reply src/main/java/model/User.java]: rebuilt project. fixed.

File: src/main/java/controller/InitServlet.java [hunk 8]
```diff
@@ -21,12 +21,21 @@
 
     @Override
     public void init() throws ServletException {
-        User user = new User("test@test", "test", "admin");
-        userService.addUser(user);
-        logger.warn("Add initial user " + user + " to DB");
+        User userFirstTest = new User("test@test", "test", "admin");
```
Comment (Sofasmile) [inline src/main/java/controller/InitServlet.java:24]: better:
firstUserTest)

File: src/main/java/controller/InitServlet.java [hunk 9]
```diff
@@ -21,12 +21,21 @@
...
+        User userSecondTest = new User("igorprihodkoemail@gmail.com",
+                "111", "user");
+        userService.addUser(userFirstTest);
+        logger.warn("Add initial user with ADMIN role " + userFirstTest + " to DB");
```
Comment (Sofasmile) [inline src/main/java/controller/InitServlet.java:28]: I think better write logger.info() here

File: src/main/java/controller/admin/EditProductServlet.java [hunk 10]
```diff
@@ -57,13 +57,13 @@ protected void doPost(HttpServletRequest req, HttpServletResponse resp)
             } else {
                 req.setAttribute("price", price);
             }
-            req.getRequestDispatcher("editProduct.jsp").forward(req, resp);
-            resp.sendRedirect("/editProduct");
+            req.getRequestDispatcher("/editProduct.jsp").forward(req, resp);
+            resp.sendRedirect("/admin/editProduct");
         }
         product.setTitle(title);
```
Comment (Sofasmile) [inline src/main/java/controller/admin/EditProductServlet.java:63]: in next hw try to create separate method for updating entities

**Review by Sofasmile — APPROVED** [review]
make small fixes)

**Pull request merged**